lib/liquid.rb

```diff
@@ -35,7 +35,7 @@ module Liquid
   QuotedFragment              = /#{QuotedString}|(?:[^\s,\|'"]|#{QuotedString})+/o
   TagAttributes               = /(\w+)\s*\:\s*(#{QuotedFragment})/o
   AnyStartingTag              = /\{\{|\{\%/
-  PartialTemplateParser       = /#{TagStart}.*?#{TagEnd}|#{VariableStart}.*?#{VariableIncompleteEnd}/om
+  PartialTemplateParser       = /#{TagStart}.*?#{TagEnd}|#{VariableStart}(?:(?:[^'"{}]+|#{QuotedString})*?|.*?)#{VariableIncompleteEnd}/om
```

Comment:
You mean it could be `(?:#{QuotedString})*?|.*?)`?  No, that would only save a quoted string at the beginning of a variable block (which normally begins with a variable name) and the following `.*?` would stop at the next `}` that might be in the next quoted string.

I included `{}` in the character class because it looked like the tokenizer has to be generous with curly brace mismatch for the "lax" parser to work.